student-store-ui/src/components/App/App.jsx

```diff
@@ -1,21 +1,133 @@
 import * as React from "react"
-import { BrowserRouter } from 'react-router-dom'
+import { useEffect, useState } from "react"
+import axios from "axios"
+
+
+
+import { BrowserRouter, Route, Routes } from 'react-router-dom'
 import Navbar from "../Navbar/Navbar"
 import Sidebar from "../Sidebar/Sidebar"
 import Home from "../Home/Home"
 import "./App.css"
+import Hero from "../Hero/Hero"
+import About from "../About/About"
+import Contact from "../Contact/Contact"
+import Footer from "../Footer/Footer"
+import ProductDetails from "../ProductDetails/ProductDetails"
 
 export default function App() {
+
+  const URL = "http://localhost:3001";
+
+  const [products, setProducts] = useState();
+  const [sidebarState, setSidebar] = useState(false);
+  const [count, setCount] = useState({});
+  const [shoppingCart, setShoppingCart] = useState([]);
+    
+  function handleAdd(e, cardId, info){
+      if(cardId in count){
+          const add = {...count, [cardId]: count[cardId] +1};
+          setCount(add);
+      }else{
+          const create = {...count, [cardId]:1};
+          setCount(create);
+      }
+
+      const cartItemm = shoppingCart?.find(item => item.id === cardId)
+
+      if (cartItemm){
+
+        const updateCart = shoppingCart?.map(item =>{
+          if(item.id === cardId){
+            return {...item, quantity:item.quantity+1}
+          }
+          return item
+        })
+
+        setShoppingCart(updateCart);
+      } else{
+        setShoppingCart( [...shoppingCart,{...info, quantity:1}])
+      }
+
+  }
+
```

Comment:
consider adding comments or creating a helper function here to make code more readable